### PR TITLE
Add repository create method

### PR DESCRIPTION
**Pull request by cspath1**
Addresses Issue #2 

- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)

### CODE REVIEW

File: src/AndcultureCode.CSharp.Sitefinity.Conductors/Extensions/DynamicContentExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+using Telerik.Sitefinity.GenericContent.Model;
+using Telerik.Sitefinity.Model;
+using Telerik.Sitefinity.RelatedData;
+using Telerik.Sitefinity.Utilities.TypeConverters;
+
+namespace AndcultureCode.CSharp.Sitefinity.Conductors.Extensions
+{
+    public static class DynamicContentExtensions
+    {
+        public static DynamicContent SetDataItemProperties<T>(this T item, PropertyInfo[] properties, string providerName) where T : SitefinityContent
```
Comment (Stefanie899) [inline src/AndcultureCode.CSharp.Sitefinity.Conductors/Extensions/DynamicContentExtensions.cs:29]: I think we should add summary comments to the extension methods in this class describing what each of them do.

File: src/AndcultureCode.CSharp.Sitefinity.Conductors/Extensions/DynamicContentExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+        private static void SetPropertyInDataItem<T>(
+           T item,
+           SitefinityMetadataAttribute metadataAttribute,
+           ref DynamicContent dataItem,
+           PropertyInfo property,
+           string providerName
+       )
+        {
+            if (!property.PropertyType.IsSubclassOf(typeof(SitefinityContent)) &&
```
Comment (Stefanie899) [inline src/AndcultureCode.CSharp.Sitefinity.Conductors/Extensions/DynamicContentExtensions.cs]: Minor: Persist the `IsSubclassOf` results in variables for readability, and so you don't have to write the same function call on line 65.

File: src/AndcultureCode.CSharp.Sitefinity.Conductors/Extensions/DynamicContentExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            string providerName
+        )
+        {
+            var parentDynamicContentType = metadataAttribute.ParentDynamicContentType;
+            var sitefinityContentItem = (SitefinityContent)item.GetType().GetProperty(property.Name)?.GetValue(item);
+
+            SitefinityMetadataAttribute sitefinityType = (SitefinityMetadataAttribute)sitefinityContentItem?.GetType()
+                .GetCustomAttribute(typeof(SitefinityMetadataAttribute));
+
+            if (sitefinityType != null)
```
Comment (Stefanie899) [inline src/AndcultureCode.CSharp.Sitefinity.Conductors/Extensions/DynamicContentExtensions.cs]: Invert the null to reduce nesting:

```
if (sitefinityType == null)
{
   return dataItem;
}
```

File: src/AndcultureCode.CSharp.Sitefinity.Conductors/Domain/DynamicContentRepositoryConductor.cs [hunk 4]
```diff
@@ -48,6 +53,21 @@ ILogger<IDynamicContentRepositoryConductor> logger
...
         }
 
+        public IResult<DynamicContent> Create<T>(T item) where T : SitefinityContent => Do<DynamicContent>.Try((r) =>
+        {
+            var type = typeof(T);
+            var properties = type.GetProperties(BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly);
+
+            var dataItem = item.SetDataItemProperties(properties, DefaultProviderName);
+
+            // TODO: shouldn't this be done in the constructor?
```
Comment (Stefanie899) [inline src/AndcultureCode.CSharp.Sitefinity.Conductors/Domain/DynamicContentRepositoryConductor.cs]: Minor: Remove or resolve `TODO`. In this case, I don't think it should be done in the constructor as it's possible the `DefaultProviderName` is changed by the user prior to the call but after the instantiation of the conductor, in order to retrieve from a different provider.

**Review by Stefanie899 — CHANGES_REQUESTED** [review]
The most pressing change prior to approval is the addition of comments to the extension methods, in my opinion.

**Review by Stefanie899 — APPROVED** [review]
Looks good now! Thanks!

**Review by KevinBusch — APPROVED** [review]

**Review by KevinBusch — APPROVED** [review]

**Review by KevinBusch — APPROVED** [review]

**Comment by brandongregoryscott** [comment]
@all-contributors add @cspath1 for code

**Comment by brandongregoryscott** [comment]
@all-contributors add @Stefanie899 for review

**Pull request merged**